src/ProducerEventProcessor/ProducerEventProcessorTest.cpp

```diff
@@ -1321,33 +1324,189 @@ TEST(ProducerEventProcessor, ThreadNameSmoke) {
   EXPECT_EQ(event.thread_name().name(), "Main Thread");
 }
 
+::testing::Matcher<ThreadStateSlice> ThreadStateSliceEq(const ThreadStateSlice& expected) {
+  return ::testing::AllOf(
+      ::testing::Property("tid", &ThreadStateSlice::tid, expected.tid()),
+      ::testing::Property("thread_state", &ThreadStateSlice::thread_state, expected.thread_state()),
+      ::testing::Property("duration_ns", &ThreadStateSlice::duration_ns, expected.duration_ns()),
+      ::testing::Property("end_timestamp_ns", &ThreadStateSlice::end_timestamp_ns,
+                          expected.end_timestamp_ns()),
+      ::testing::Property("wakeup_reason", &ThreadStateSlice::wakeup_reason,
+                          expected.wakeup_reason()),
+      ::testing::Property("wakeup_tid", &ThreadStateSlice::wakeup_tid, expected.wakeup_tid()),
+      ::testing::Property("wakeup_pid", &ThreadStateSlice::wakeup_pid, expected.wakeup_pid()),
+      ::testing::Property("switch_out_or_wakeup_callstack_status",
+                          &ThreadStateSlice::switch_out_or_wakeup_callstack_status,
+                          expected.switch_out_or_wakeup_callstack_status()),
+      ::testing::Property("switch_out_or_wakeup_callstack_id",
+                          &ThreadStateSlice::switch_out_or_wakeup_callstack_id,
+                          expected.switch_out_or_wakeup_callstack_id()));
+}
+
+::testing::Matcher<Callstack> CallstackEq(const Callstack& expected) {
+  return ::testing::AllOf(
+      ::testing::Property("pcs", &Callstack::pcs, ElementsAreArray(expected.pcs())),
+      ::testing::Property("type", &Callstack::type, expected.type()));
+}
+
+::testing::Matcher<InternedCallstack> InternedCallstackEq(const InternedCallstack& expected) {
+  return ::testing::AllOf(
+      ::testing::Property("key", &InternedCallstack::key, expected.key()),
+      ::testing::Property("intern", &InternedCallstack::intern, CallstackEq(expected.intern())));
+}
+
+::testing::Matcher<ClientCaptureEvent> ClientCaptureEventsTheadStateSliceEq(
+    const ThreadStateSlice& expected_thread_state_slice) {
+  return ::testing::Property("thread_state_slice", &ClientCaptureEvent::thread_state_slice,
```

Comment:
I think we should also check that `has_thread_state_slice` is true. Similarly in `ClientCaptureEventsInternedCallstackEq`.

Reply:
Done.